### ISSUE TITLE
Tracking issue for `soft_unstable` compatibility lint

### ISSUE THREAD

**Issue by petrochenkov** [rust-lang/rust#64266]
#### What is this lint about

When some feature is unstable, but was allowed on stable due to some stability checking hole or oversight we may unstabilize it in a "soft" way to avoid breaking crates depending on the crates using the feature.

This soft unstabilization can be done using this lint.

#### How to fix this warning/error

Remove or `cfg`-out uses of the unstable feature in configurations of your code that are supposed to build on stable.

#### Current status
- https://github.com/rust-lang/rust/pull/64066 uses `soft_unstable` for the attribute `#[bench]`.

**Comment by lqd** [comment]
`#[deny(soft_unstable)]` is on by default, but the error message ([playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f24c5d070928af2b2c38e8e55b6757da)) still has a lint-like note warning mentioning it will become a hard error in the future:
```
= warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
= note: for more information, see issue #64266 <https://github.com/rust-lang/rust/issues/64266>
```

 do we need an issue for this ?

**Comment by Stargateur** [comment]
So this feature is used to make breaking change to avoid breaking change ? The concept of `solf_unstable` don't make any sense for me, specially `#[deny(soft_unstable)]` if is default. It's just more complicate way that produce the same result. Or I'm missing something.

`#[stability(level=2.33333333, wind-strenght=5.3)]` (not from me, author probably wish to stay in shadow)

**Comment by yaahc** [comment]
I think I might need to use this for https://github.com/rust-lang/rust/pull/95956/ as well

**Comment by kryptan** [comment]
Why doesn't compiler error message say what feature should I enable? It tells me about the lint `soft_unstable` but not about `feature(custom_inner_attributes)`. Error messages for unstable features typically tell you the name of the feature.
